python/cuml/cuml/cluster/kmeans.pyx

```diff
@@ -302,6 +306,11 @@ class KMeans(UniversalBase,
                                                   else None),
                                 check_dtype=check_dtype)
 
+        # XXX Should deriving a seed from a random state be idempotent? Should repeated
+        # XXX calls of `fit` create new seeds or not?
```

Comment:
What do people think about this? Should we re-derive a seed each time `fit` is called?

Reply:
If you pass an `int` each call to `fit` is the same, but if you pass a random state it keeps getting forwarded, so each `fit` is different. (I think it is at least somewhat unclear what should happen, at least within scikit-learn we've not really been able to converge on something :-/)

I think here I'd vote for deriving a new seed each time. My thinking is that that way we match scikit-learn (no need to somehow special case this for the accelerator). Even if I can't justify why having a new seed each time is "the right thing to do"